tests/unit_test.py

```diff
@@ -75,3 +75,90 @@ def test_default_unit_for_string_is_none():
                                sc.Unit('12.3 * m/A*kg^2/rad^3'), sc.Unit('count')))
 def test_dict_roundtrip(u):
     assert units_identical(sc.Unit.from_dict(u.to_dict()), u)
+
+
+@pytest.fixture
+def clean_unit_aliases():
+    sc.units.clear_unit_aliases()
+    yield
+    sc.units.clear_unit_aliases()
+
+
+def test_unit_alias_overrides_to_string(clean_unit_aliases):
+    sc.units.add_unit_alias(name='clucks', unit='19.3 m*A')
+    clucks = sc.Unit('19.3 m*A')
+    assert str(clucks) == 'clucks'
+    assert str(sc.Unit('one') / clucks) == '1/clucks'
+    assert str(clucks**2) == 'clucks^2'
+    assert str(clucks * sc.Unit('kg')) == 'clucks*kg'
+
+
+def test_can_add_multiple_aliases(clean_unit_aliases):
+    sc.units.add_unit_alias(name='clucks', unit='19.3 m*A')
+    sc.units.add_unit_alias(name='dogyear', unit='4492800s')
+    assert str(sc.Unit('4492800s')) == 'dogyear'
+    assert str(sc.Unit('19.3 m*A')) == 'clucks'
+
+
+def test_unit_alias_enables_conversion_from_string(clean_unit_aliases):
+    sc.units.add_unit_alias(name='speed', unit='m/s')
+    assert sc.Unit('speed') == sc.Unit('m/s')
+    assert sc.Unit('1/speed') == sc.Unit('s/m')
+    assert sc.Unit('speed/K') == sc.Unit('m/s/K')
+
+
+def test_can_remove_unit_alias(clean_unit_aliases):
+    sc.units.add_unit_alias(name='clucks', unit='19.3 m*A')
+    sc.units.add_unit_alias(name='dogyear', unit='4492800s')
+
+    sc.units.remove_unit_alias(name='dogyear')
+    assert str(sc.Unit('19.3 m*A')) == 'clucks'
+    assert 'dogyear' not in str(sc.Unit('4492800s'))
+    with pytest.raises(sc.UnitError):
+        sc.Unit('dogyear')
+
+    sc.units.remove_unit_alias(name='clucks')
+    assert 'clucks' not in str(sc.Unit('19.3 m*A'))
+    assert 'dogyear' not in str(sc.Unit('4492800s'))
+    with pytest.raises(sc.UnitError):
+        sc.Unit('dogyear')
+    with pytest.raises(sc.UnitError):
+        sc.Unit('clucks')
+
+
+def test_clear_unit_alias(clean_unit_aliases):
+    sc.units.add_unit_alias(name='speed', unit='m/s')
+    sc.units.add_unit_alias(name='chubby', unit='100kg')
+
+    sc.units.clear_unit_aliases()
+    assert 'speed' not in str(sc.Unit('m/s'))
+    assert 'chubby' not in str(sc.Unit('100kg'))
+    with pytest.raises(sc.UnitError):
+        sc.Unit('speed')
+    with pytest.raises(sc.UnitError):
+        sc.Unit('chubby')
+
+
+def test_removing_undefined_alias_does_nothing(clean_unit_aliases):
+    sc.units.add_unit_alias(name='chubby', unit='100kg')
+
+    sc.units.remove_unit_alias(name='clucks')
```

Comment:
Shouldn't this raise?

Reply:
The idea was that a user would declare that they don't want a certain unit to show up. So if it wasn't even declared, removing it would just be a noop. But I see your point. And with a dict-syntax, it should definitely raise.

Also, it might be confusing. If, e.g., a user removes 'barn', it does not complain but strings will still contain it because it is built in.

Reply:
> The idea was that a user would declare that they don't want a certain unit to show up.

A `pop` method could complement the `__delitem__` for this purpose.